roles/ceph-common/tasks/checks/check_mandatory_vars.yml

```diff
@@ -23,26 +27,36 @@
     ceph_stable_rh_storage and
     not ceph_stable_rh_storage_cdn_install and
     not ceph_stable_rh_storage_iso_install
+  tags:
+    - package-install
 
 - name: make sure journal_size configured
   fail:
     msg: "journal_size must be configured. See http://ceph.com/docs/master/rados/configuration/osd-config-ref/"
-  when: journal_size|int == 0
+  when:
+    journal_size|int == 0 and
+    osd_group_name in group_names
```

Comment:
Monitors do not need to have the osd conf to work but it is better to have the osd settings from the get go.
The only thing that all the daemons share in common is the `ceph.conf`, so if you discover some setting after the monitors have been bootstrapped, the file will change and then handlers will get triggered which will issue an unnecessary restart of the monitors. Perhaps if you use `--limit` while running the playbook you can avoid that but you will get inconsistent `ceph.conf` between monitors and osds.

How difficult is it to add these options even on the mons bootstrap?